File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        ctMap = defaultdict(int)
        for num in nums:
            ctMap[num] += 1
        ctMapSorted = dict(sorted(ctMap.items(), key=lambda item: item[1], reverse=True))
        res = []
        for key in ctMapSorted.keys():
            if k == 0:
                return res
            res.append(key)
            k -= 1
        return res
        #WCRT: O(NlogN) | Space: O(N)

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> hashmap;
        for (int num : nums){
            if (hashmap[num] == 1){
                return true;
            }
            hashmap[num] = 1;
        }
        return false;
    }
};